sdk/src/main/java/ch/admin/bag/covidcertificate/eval/net/JwsKeyResolver.kt

```diff
@@ -0,0 +1,47 @@
+package ch.admin.bag.covidcertificate.eval.net
+
+import android.util.Base64
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.JwsHeader
+import io.jsonwebtoken.SigningKeyResolverAdapter
+import java.io.ByteArrayInputStream
+import java.lang.Exception
+import java.security.Key
+import io.jsonwebtoken.security.SignatureException
+import java.security.cert.CertificateFactory
+
+import java.security.cert.X509Certificate
+
+ class JwsKeyResolver(private val rootCA: X509Certificate) : SigningKeyResolverAdapter() {
+
+    override fun resolveSigningKey(jwsHeader: JwsHeader<*>, claims: Claims) : Key
+     {
+         val encodedCertificates : List<String> = jwsHeader["x5c"] as List<String>? ?: throw SignatureException("JWS is missing the required certificate chain")
+        var certificates : MutableList<X509Certificate> = mutableListOf()
+        val certificateFactory = CertificateFactory.getInstance("X.509")
+        for ( cert in encodedCertificates) {
+            val certBytes = Base64.decode(cert, Base64.DEFAULT)
+            val byteInputStream = ByteArrayInputStream(certBytes)
+            val x509 = certificateFactory.generateCertificate(byteInputStream) as X509Certificate
```

Comment:
This could throw a CertificateException on parsing errors - do we trust the backend enough not to catch them?

Reply:
Good point, we should catch it and throw a `SignatureException` as well (which will be handled by the framework).

Reply:
is this resolved?